### PR TITLE
Core/Misc: Implement Chromie opcodes

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Implement Chromie opcodes

**Tests performed:**

Tested in-game

### CODE REVIEW

**Comment by MaxtorCoder** [comment]
Move this to the other PR please :)

File: src/server/game/Entities/Player/Player.h [hunk 1]
```diff
@@ -2761,6 +2761,8 @@ class TC_GAME_API Player : public Unit, public GridObject<Player>
 
         void SetTransportServerTime(int32 transportServerTime) { SetUpdateFieldValue(m_values.ModifyValue(&Player::m_activePlayerData).ModifyValue(&UF::ActivePlayerData::TransportServerTime), transportServerTime); }
 
+        void SetChromieTimeExpansion(uint8 chromieTimeExpansion) { SetUpdateFieldValue(m_values.ModifyValue(&Player::m_activePlayerData).ModifyValue(&UF::ActivePlayerData::UiChromieTimeExpansionID), chromieTimeExpansion); }
```
Comment (fluxurion) [inline src/server/game/Entities/Player/Player.h:2764]: This function can send ExpansionLevelMask too with a db2 function to get chromie time expansion info where the ID = chromieTimeExpansion argument of the function

File: src/server/game/Spells/SpellEffects.cpp [hunk 2]
```diff
@@ -5919,3 +5919,16 @@ void Spell::EffectLearnTransmogIllusion()
...
+void Spell::EffectSetChromieTime()
+{
+    if (effectHandleMode != SPELL_EFFECT_HANDLE_HIT_TARGET)
+        return;
+
+    Player* caster = m_caster->ToPlayer();
+    if (!caster)
+        return;
+
+    UiChromieTimeExpansionInfoEntry const* expansion = sUiChromieTimeExpansionInfoStore.LookupEntry(effectInfo->MiscValue);
```
Comment (MaxtorCoder) [inline src/server/game/Spells/SpellEffects.cpp:5932]:
```suggestion
    UiChromieTimeExpansionInfoEntry const* expansion = sUiChromieTimeExpansionInfoStore.LookupEntry(effectInfo->MiscValue);
    if (!expansion)
        return;
```

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellEffects.cpp (hunk 2 above)
Comment (aquadeus) [reply src/server/game/Spells/SpellEffects.cpp]: thanks for reminding me to handle select the present xd

**Pull request closed without merging**